Sources/Datadog/DatadogConfiguration.swift

```diff
@@ -96,28 +108,47 @@ extension Datadog {
 
         /// `Datadog.Configuration` builder.
         ///
-        /// Usage:
+        /// Usage (to enable RUM, Logging and Tracing):
+        ///
+        ///     Datadog.Configuration.builderUsing(rumApplicationID:rumClientToken:environment:)
+        ///                           ... // customize using builder methods
+        ///                          .build()
+        ///
+        /// or (to only enable Logging and Tracing):
         ///
-        ///     Datadog.Configuration.builderUsing(clientToken: "<client token>", environment: "<env name>")
+        ///     Datadog.Configuration.builderUsing(clientToken:environment:)
         ///                           ... // customize using builder methods
         ///                          .build()
         ///
         public class Builder {
-            internal let clientToken: String
-            internal let environment: String
-            internal var loggingEnabled = true
-            internal var tracingEnabled = true
-            internal var rumEnabled = true
-            internal var logsEndpoint: LogsEndpoint = .us
-            internal var tracesEndpoint: TracesEndpoint = .us
-            internal var rumEndpoint: RUMEndpoint = .us
-            internal var serviceName: String? = nil
-            internal var tracedHosts = Set<String>()
-            internal var rumSessionsSamplingRate: Float = 100.0
+            internal var configuration: Configuration
+
+            /// Initializer invoked when initializing the SDK to use any of (including all): RUM, Logging, Tracing.
+            internal convenience init(rumApplicationID: String, rumClientToken: String, environment: String) {
```

Comment:
the other approach I had in mind is to make those constructors private and provide 2 static methods:
```
builderWith(rumApplicationId, rumClientToken, environment): DefaultBuilder
builderWith(rumApplicationId, rumClientToken, environment): RumBuilder

```

And then you would have:

```
class RumBuilder: DefaultBuilder {
     enableRum(enabled:Booleant)
     ....
}
```

This approach will avoid by default the situation where the client enables RUM but he's not adding the AppID but will mean modifying the Public API :(

Reply:
Yes 👌, I've been considering this as well, but decided that this can come after, as the improvement (also, we have an idea of replacing the builder pattern with simple `struct`). Anyway, this won't change the nature of `FeaturesConfiguration`, as we always have to validate the user configuration.